### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by jplatte**

### CODE REVIEW

**Review by Peternator7 — CHANGES_REQUESTED** [review]
If we are going to switch to `Result` style error handling we should try and switch all the `panic`s to return `Err` instead.

Otherwise, everything looks great.

File: strum_macros/src/macros/enum_iter.rs [hunk 1]
```diff
@@ -24,17 +25,19 @@ pub fn enum_iter_inner(ast: &syn::DeriveInput) -> TokenStream {
     };
 
     let variants = match ast.data {
-        syn::Data::Enum(ref v) => &v.variants,
+        Data::Enum(ref v) => &v.variants,
         _ => panic!("EnumIter only works on Enums"),
```
[the remark this answers is not in the document]
Comment (Peternator7) [reply strum_macros/src/macros/enum_iter.rs:29]: Perhaps we can use [`Span.call_site()`](https://docs.rs/proc-macro2/1.0.19/proc_macro2/struct.Span.html#method.call_site) as the error position here.

**Comment by jplatte** [comment]
Yes, the plan is to convert more panics to `syn::Error`. This is just the start :)

**Review by Peternator7 — APPROVED** [review]

**Comment by Peternator7** [comment]
Change is merged. I'm going to hold off publishing a new version until the rest of your changes are in.

**Pull request merged**